mozilla/addons-server
# Convert block type to tiny integer with enum supporting field type

Fixes: mozilla/addons#15111


### Description

This PR achieves 3 goals:
- add a `TinyIntegerField` class for storing 1 bit integers in mysql
- migrates the `BlockVersion.soft` field to `BlockVersion.block_type` using that field
- migrates the `BlockListSubmission.block_type` field to using that field as well

### Context

It was somewhat clear at the beginning of introducing soft blocking that a boolean field might not be totally adequate for our needs.

Though we did not want to introduce larger size integer fields at the time. It is now clear we should do this. It makes the ergonomics of using choices for the types of blocks much easier and since we added the tinyint field it is equivalent in space efficiency to a boolean.

### Testing

It should be sufficient for automated tests on this one but you can test the migration by

- create blocklistsubmission and blockversion records (both blocked and soft_blocked)
- run the migrations (make up)
- check the database preserves existing values.
- bonus: you should be able to roll back the migration reintroducing the soft field with correct values from before

Additionally you could attempt creating blocked/soft_blocked versions via admin and api and ensure no regressions.

### Checklist
